src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx

```diff
@@ -196,19 +199,62 @@ export const TradeConfirm = () => {
     if (sellTxHash) return getSellTxLink(sellTxHash)
   }, [buyTxHash, getBuyTxLink, getSellTxLink, sellTxHash])
 
-  const { shapeShiftFee, donationAmount } = useMemo(() => {
-    if (applyThorSwapAffiliateFees && swapperName === SwapperName.Thorchain && tradeQuote) {
-      return {
-        shapeShiftFee: {
-          amountFiatPrecision: quoteAffiliateFeeFiatPrecision ?? '0',
-          amountBps: tradeQuote.affiliateBps ?? '0',
-        },
-        donationAmount: undefined,
+  const _donationAmountUserCurrency = useAppSelector(selectQuoteDonationAmountUserCurrency)
+  const potentialDonationAmountUserCurrency = useAppSelector(
+    selectPotentialDonationAmountUserCurrency,
+  )
+  const potentialAffiliateBps = useAppSelector(selectActiveQuotePotentialDonationBps)
+  const affiliateBps = useAppSelector(selectActiveQuoteAffiliateBps)
+
+  const { shapeShiftFee, donationAmountUserCurrency } = useMemo(() => {
+    if (tradeQuote) {
+      if (isFoxDiscountsEnabled) {
+        const feeDiscountUserCurrency = bnOrZero(potentialDonationAmountUserCurrency)
+          .minus(_donationAmountUserCurrency)
+          .toString()
+        return {
+          shapeShiftFee: {
+            amountAfterDiscountUserCurrency: _donationAmountUserCurrency ?? '0',
+            amountBeforeDiscountUserCurrency: potentialDonationAmountUserCurrency ?? '0',
+            feeDiscountUserCurrency,
+            affiliateBps: affiliateBps ?? '0',
+            potentialAffiliateBps: potentialAffiliateBps ?? '0',
+            foxDiscountPercent: bnOrZero(feeDiscountUserCurrency)
+              .div(potentialDonationAmountUserCurrency ?? 0)
+              .toString(),
+          },
+        }
+      } else {
+        // The donation/shapeshiftFee vernacular is weird but expected for THOR, see https://github.com/shapeshift/web/pull/5230
+        if (applyThorSwapAffiliateFees && swapperName === SwapperName.Thorchain && tradeQuote) {
+          return {
+            shapeshiftFee: {
+              amountAfterDiscountUserCurrency: potentialDonationAmountUserCurrency ?? '0',
+              amountBeforeDiscountUserCurrency: potentialDonationAmountUserCurrency ?? '0',
+              amountBps: tradeQuote.potentialAffiliateBps ?? '0',
+            },
+            donationAmountUserCurrency: undefined,
+          }
+        }
+
+        return { shapeShiftFee: undefined, donationAmountUserCurrency: _donationAmountUserCurrency }
```

Comment:
can this whole function be broken into a helper and re-used in `TradeInput`?

Reply:
9ab7a9b199dfc7e17aaa2aa0606f9e18001f46c9 and tackled types while at it, this will be *actually* safe now